compiler/rustc_mir_transform/src/known_panics_lint.rs

```diff
@@ -897,8 +885,8 @@ impl CanConstProp {
         for (local, val) in cpv.can_const_prop.iter_enumerated_mut() {
             let ty = body.local_decls[local].ty;
             if ty.is_union() {
-                // Do not const prop unions as they can
-                // ICE during layout calc
+                // Unions are incompatible with const prop because
```

Comment:
Incompatible with this simple const prop impl

It's not a fundamental issue with const propagators

Reply:
Yeah, I meant that. Should've been more explicit. Will update it.

Reply:
Done